### ISSUE TITLE
Mobile Module: phone portrait upload goes wrong when viewing on the full site

### ISSUE THREAD

**Issue by Mthorganby** [SocialEngine/phpv4-issues#720]
PHP 4.91 full install, standard settings, straight out the box
Using Google Chrome as browser
-------------------------

1. Take a photo with a mobile in portrait mode.
2. Use you mobile to upload to a photo album/activity feed.

In the background your mobile phone ads an IMAGE EXIF of Orientation: **Rotate 90 CW** when saving the image.

3. The upload works!

When you view the photo on your mobile, you see a nice portrait photo.

Looking on the full website at the photo, it is displayed in landscape:  **90CCW**.

Where it seems to go wrong:

You mobile phone ads an IMAGE EXIF of Orientation: Rotate 90 CW when saving the image.
Your mobile phone is not fooled hereby. Reads it how it comes, if rotation is needed it will do so.

But on the full website your browser doesn't seem to reed the Orientation EXIF: Rotate 90 CW
And processes the photo as a landscape image, ignoring the EXIF full stop.

See sample:
The top small image is how it looks like on the full website, even though it show portait on the mobile phone.

PS: Don't laugh I still use an iPhone 4S, it is still going strong!! :)

![mobile-1](https://cloud.githubusercontent.com/assets/25981420/25557603/c942e4ce-2d0c-11e7-859b-21519cbf6c26.png)

**Comment by DonnaScriptTechs** [comment]
Please make sure you are using 4.9.1 and not 4.9.0. We had a release 4.9.1 which had fixed issues. The patch p1 was a different thing.

**Comment by Mthorganby** [comment]
@DonnaScriptTechs sorry made a spelling mistake, it is the latest version SE PHP 4.91 have updated the post

**Comment by DonnaScriptTechs** [comment]
I'll check with the core developers on Monday. We might need to do as other scripts do and put a list of supported browsers/os/devices. I don't know atm how old devices work or if they are supported but will find out when I'm working. :)

**Comment by Mthorganby** [comment]
Thanks

**Comment by gsf00001** [comment]
@Mthorganby ==> "PS: Don't laugh I still use an iPhone 4S, it is still going strong!! :)  "
I'm smiling, not laughing - I use a 2012 Droid4 because I must have a physical kbd.  I also have 2 fresh new backups from ebay that I update and charge once a month.  So I may be 'worse' than you :)

**Comment by Elshara** [comment]
I love blue tooth keyboards. They serve me well I love the older
phones myself. Samsung 5s and earlier. Iphone 6s and earlier. The
pixil is my favorite however and so I use it with the keyboard. Works
well. Otherwise I'm always on my asus Windows 10 laptop. I love
Android 7.2 nougat.

**Comment by DonnaScriptTechs** [comment]
I'm able to reproduce this with an iphone 5c. I marked it a bug but a core developer needs to check as I'm not sure if it's a bug or configuration issue or older phone issue as my phone is old too.

**Comment by Mthorganby** [comment]
Hi Donna, if you are running the latest iOS, it should use the latest EXIF DATA version to write to files

**Comment by Mthorganby** [comment]
I have spoken with SEAO and they say it is a problem within SE core

**Comment by DonnaScriptTechs** [comment]
@Mthorganby I'm not sure why you keep updating this as I marked it a bug when I mentioned I can reproduce it. No need to keep testing and updating as we have it marked a bug. Thank you! :)

**Comment by DonnaScriptTechs** [comment]
Our core developers will check it and determine further. Thanks for your contribution.

**Comment by poeticjustice** [comment]
I just want to add something here,
I experienced similar issue, and it was fonud to be because of capitals.

iPhones save photos as either .JPEG or .JPG

If the file is .jpeg .jpg the auto rotate feature worked perfectly... note this is 4.8.12 when i personally fixed an issue.

but uppercase files had many issues. 

May help testing

**Comment by Mthorganby** [comment]
@poeticjustice  I had a problem with that with an online auction site, it took ages to figure that problem out. I spotted it after a week or two why they weren't uploading and not showing at all. 
But this is Linux and Microshit for you!
I had to rewrite some of the code, so it was writing the whole file name to lowercase. And it solved the problem. The HTML was putting lowercase out and some file had uppercase extensions which resulted in showing the wrong image location or an non-existing image file.
This could be the case here
